Framework/src/HashDataDescription.cxx

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019-2024 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <iostream>
+#include <sstream>
+
+#include "QualityControl/HashDataDescription.h"
+
+namespace o2::common::hash
+{
+
+// creates hash of input string and returns hexadecimal representation
+auto to_hexa(const std::string& input, size_t hash_length) -> std::string
+{
+  std::stringstream ss;
+  ss << std::hex << std::hash<std::string>{}(input);
+  return std::move(ss).str().substr(0, hash_length);
+};
+
+o2::header::DataDescription createDataDescription(const std::string& name, size_t hashLength)
+{
+  o2::header::DataDescription description{};
+  if (name.size() <= o2::header::DataDescription::size) {
+    description.runtimeInit(name.c_str());
+    return description;
+  } else {
+
+    std::stringstream ss{};
+    ss << std::hex << std::hash<std::string>{}(name);
```

Comment:
Perhaps mac failures are related to the implementation of `std::hash`? I think it would be good to have the hash to return the same result on any platform and any execution, because we might generate a workflow template on one OS and use it in a different one.

Reply:
yes, it does... I thought that it might happen, but I still wanted to try it

Reply:
I had a implementation of djb2 hash, but I too it away when I saw that `std::hash` is used for timers.

Reply:
then indeed it's weird that it does not fail for timers...

Reply:
https://github.com/AliceO2Group/QualityControl/blob/master/Framework/src/TaskRunner.cxx#L307-L320

I guess that there is not a test for timers, that tests for exact task name